File: server/Models/Voyage.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace server.Models;

public class Voyage
{
    [Key]
    public int IdVoyage { get; set; }

    [Required]
    public DateTime VoyageDate { get; set; }

    [Required]
    public DateTime StartTime { get; set; }

    [Required]
    public DateTime EndTime { get; set; }

    //FK
    public int IdShip { get; set; }
    public Ship Ship { get; set; } = null!;

    public int DeparturePortId { get; set; }
    public Port DeparturePort { get; set; } = null!;

    public int ArrivalPortId { get; set; }
    public Port ArrivalPort { get; set; } = null!;
}

File: client/src/app/models/voyage.ts
export interface VoyageRequestDto {
  idShip: number;
  departurePortId: number;
  arrivalPortId: number;
  voyageDate: string;
  startTime: string;
  endTime: string;
}

export interface VoyageResponseDto {
  idVoyage: number;
  idShip: number;
  shipName: string;
  departurePortId: number;
  departurePortName: string;
  arrivalPortId: number;
  arrivalPortName: string;
  voyageDate: string;
  startTime: string;
  endTime: string;
}
